components/style/selector_matching.rs

```diff
@@ -373,7 +369,8 @@ impl Stylist {
         map.user_agent.normal.get_all_matching_rules(element,
                                                      parent_bf,
                                                      applicable_declarations,
-                                                     &mut relations);
```

Comment:
Maybe, since each map contains rules for a given importance, the importance could be stored on the map itself?